### PR TITLE
LAK-1039: Introspect table names in draft datasource config action

### PR DESCRIPTION
**Pull request by roblim**
- Updated DraftDatasourceConfigAction  to introspect table names and update DraftConfig

### CODE REVIEW

**Comment by roblim** [comment]
PR is dependent on https://github.com/greatexpectationslabs/mercury/pull/3386

File: poetry.lock
Comment (Kilo59) [inline poetry.lock:1]: We shouldn't need to update the lockfile.

File: great_expectations_cloud/agent/actions/draft_datasource_config_action.py [hunk 1]
```diff
@@ -58,7 +63,42 @@ def check_draft_datasource_config(
...
+
+        return ActionResult(
+            id=id,
+            type=event.type,
+            created_resources=[],
+        )
+
+    def _get_table_names(self, datasource: Datasource) -> list[str]:
+        inspector: Inspector = inspect(datasource.get_engine())
+        return inspector.get_table_names()
```
Comment (Kilo59) [inline great_expectations_cloud/agent/actions/draft_datasource_config_action.py:79]: Correct me if something has changed but we aren't providing the `schema` unless we discover an issue?

File: great_expectations_cloud/agent/actions/draft_datasource_config_action.py [hunk 2]
```diff
@@ -58,7 +63,42 @@ def check_draft_datasource_config(
...
+            f"{cloud_config.gx_cloud_organization_id}/datasources/drafts/{config_id}",
+            json={"table_names": table_names},
+        )
+        if not response.ok:
+            raise RuntimeError(  # noqa: TRY003 # one off error
+                f"DraftDatasourceConfigAction encountered an error while connecting to GX Cloud. "
+                f"Unable to update "
+                f"table_names for Draft Config with ID"
+                f"={config_id}.",
+            )
```
Comment (Kilo59) [inline great_expectations_cloud/agent/actions/draft_datasource_config_action.py:101]: `non-blocking`
Do all of the agent jobs raise `RuntimeError`s?

File: tests/agent/actions/test_draft_datasource_config.py [hunk 3]
```diff
@@ -60,11 +60,87 @@ def test_test_draft_datasource_config_success(
...
     )
     session = create_session.return_value
-    response = session.get.return_value
-    response.ok = True
-    response.json.return_value = build_payload(config=datasource_config, id=config_id)
+    get_draft_config_response = session.get.return_value
+    get_draft_config_response.ok = True
+    get_draft_config_response.json.return_value = build_get_draft_config_payload(
+        config=datasource_config, id=config_id
+    )
```
Comment (Kilo59) [inline tests/agent/actions/test_draft_datasource_config.py:67]: Can you use `responses` for HTTP mocking.
https://github.com/getsentry/responses?tab=readme-ov-file#shortcuts

Examples from this repo.

https://github.com/great-expectations/cloud/blob/d366b02288c5068e07a557ee48e1a3c90d68bb68/tests/agent/test_agent.py#L438-L467

https://github.com/great-expectations/cloud/blob/d366b02288c5068e07a557ee48e1a3c90d68bb68/tests/agent/actions/test_list_table_names_action.py#L78-L95

**Review by Kilo59 — CHANGES_REQUESTED** [review]
Looks good.

Made a couple of suggestions. Requesting changes for the following 2 items. Consider the rest of my comments/suggestions non-blocking.

1. Drop the `poetry.lock` change.
2. Use `responses` for the HTTP mocking.

File: great_expectations_cloud/agent/actions/draft_datasource_config_action.py (hunk 1 above)
Comment (roblim) [reply great_expectations_cloud/agent/actions/draft_datasource_config_action.py:79]: Correct

File: great_expectations_cloud/agent/actions/draft_datasource_config_action.py (hunk 2 above)
Comment (roblim) [reply great_expectations_cloud/agent/actions/draft_datasource_config_action.py:101]: Yep

File: poetry.lock
Comment (roblim) [reply poetry.lock:1]: Addressed in https://github.com/great-expectations/cloud/pull/305/commits/cb99fa81e05ddadd964f384a667d05a7ab009fa2

File: tests/agent/actions/test_draft_datasource_config.py (hunk 3 above)
Comment (roblim) [reply tests/agent/actions/test_draft_datasource_config.py]: Oh man, much cleaner. Thanks for the suggestion/reminder! Addressed in https://github.com/great-expectations/cloud/pull/305/commits/e71792cce9f48bd427bfd96c35db7709d772e506

**Review by Kilo59 — APPROVED** [review]
On my phone so haven't looked thoroughly, but you might be able to refactor the responses mock to be a fixture to make it easier to reuse.

**Pull request merged**